### PR TITLE
Update win64 openssl provisioning script

### PR DESCRIPTION
**Pull request by momopranto**
Minor fixes to provisioning script:
* moved `Invoke-VcVarsAll` to `osquery_utils.ps1` so all provisioning scripts can use it
* Force set TLS 1.2 in `osquery_utils.ps1`
* Explicitly check Perl install to make sure incompatible version is not used
* Clean up external executions for things like `.bat` scripts and 7z, nmake, etc

### CODE REVIEW

File: tools/provision/chocolatey/openssl.ps1 [hunk 1]
```diff
@@ -31,7 +31,11 @@ $curlCertsShaSum =
 . "$(Split-Path -Parent $MyInvocation.MyCommand.Definition)\osquery_utils.ps1"
 
 # Invoke the MSVC developer tools/env
-Invoke-BatchFile "$env:VS140COMNTOOLS\..\..\vc\vcvarsall.bat" amd64
+$ret = Invoke-VcVarsAll
+if ($ret -ne $true) {
+	Write-Host "[-] vcvarsall.bat failed to run" Red
```
Comment (muffins) [inline tools/provision/chocolatey/openssl.ps1]: Be explicit with arguments:
```
Write-Host "[-] vcvarsall.bat failed to run" -ForegroundColor Red
```

File: tools/provision/chocolatey/openssl.ps1 [hunk 2]
```diff
@@ -88,15 +93,20 @@ if (-not (Test-Path $zipFile)) {
   Invoke-WebRequest $url -OutFile "$zipFile"
 }
 
+$7z = (Get-Command '7z').Source
```
Comment (muffins) [inline tools/provision/chocolatey/openssl.ps1]: For every invocation of `Get-Command` that you use, check the output, and ensure that we `-ErrorAction SilentlyContinue`

File: tools/provision/chocolatey/openssl.ps1 [hunk 3]
```diff
@@ -88,15 +93,20 @@ if (-not (Test-Path $zipFile)) {
...
+$7zargs = 'x ' + $zipFile
```
Comment (muffins) [inline tools/provision/chocolatey/openssl.ps1]: Make this an array so you can ditch the space:
```
$7zargs = @(
  'x',
   $zipFile
)
```

File: tools/provision/chocolatey/openssl.ps1 [hunk 4]
```diff
@@ -88,15 +93,20 @@ if (-not (Test-Path $zipFile)) {
   Invoke-WebRequest $url -OutFile "$zipFile"
 }
```
Comment (muffins) [inline tools/provision/chocolatey/openssl.ps1]: Add in logic to derive and create the build path:
```
$loc = Get-Location
# Keep the location of build script, to bring with in the chocolatey package
$buildScriptSource = $MyInvocation.MyCommand.Definition

# Create the choco build dir if needed
$buildPath = Get-OsqueryBuildPath
if ($buildPath -eq '') {
  Write-Host '[-] Failed to find source root' -foregroundcolor red
  exit
}
$chocoBuildPath = "$buildPath\chocolatey\$packageName"
if (-not (Test-Path "$chocoBuildPath")) {
  New-Item -Force -ItemType Directory -Path "$chocoBuildPath"
}
Set-Location $chocoBuildPath
```

File: tools/provision/chocolatey/openssl.ps1 [hunk 5]
```diff
@@ -88,15 +93,20 @@ if (-not (Test-Path $zipFile)) {
...
 # Build the libraries
-perl Configure VC-WIN64A
-ms\do_win64a
-nmake -f ms\nt.mak
+Start-OsqueryProcess $perl 'Configure VC-WIN64A' $false
+Invoke-BatchFile 'ms\do_win64a.bat'
+Start-OsqueryProcess $nmake '-f ms\nt.mak' $false
 
 #$buildDir = New-Item -Force -ItemType Directory -Path 'osquery-win-build'
 #Set-Location $buildDir
```
Comment (muffins) [inline tools/provision/chocolatey/openssl.ps1]: Lastly, uncomment this

File: tools/provision/chocolatey/openssl.ps1 [hunk 6]
```diff
@@ -88,15 +93,20 @@ if (-not (Test-Path $zipFile)) {
...
 
 # Build the libraries
-perl Configure VC-WIN64A
-ms\do_win64a
-nmake -f ms\nt.mak
+Start-OsqueryProcess $perl 'Configure VC-WIN64A' $false
+Invoke-BatchFile 'ms\do_win64a.bat'
+Start-OsqueryProcess $nmake '-f ms\nt.mak' $false
 
 #$buildDir = New-Item -Force -ItemType Directory -Path 'osquery-win-build'
```
Comment (muffins) [inline tools/provision/chocolatey/openssl.ps1]: Uncomment this, but take it up above before we do the `7z`, as per the comment.

File: tools/provision/chocolatey/openssl.ps1 (hunk 4 above)
Comment (momopranto) [reply tools/provision/chocolatey/openssl.ps1]: This is all already in there L80-92. Unless Im confused about what you're requesting 😕

**Pull request merged**